File: Project2/warehouse/server/src/models/request.js
const { messageType } = require('../enums');
const { sendNotificationMessage } = require('../services/websockets/pusher');
const { PUSHER_CHANNEL_WAREHOUSE } = require('../config/configs');

module.exports = (sequelize, DataTypes) => {
  const Request = sequelize.define('Request', {
    quantity: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 0,
    },
    processedDate: {
      type: DataTypes.DATE,
    },
  }, {});

  Request.associate = (models) => {
    Request.belongsTo(models.Book, {
      foreignKey: 'bookId',
    });
  };

  Request.afterCreate(async (request) => {
    const book = await request.getBook();
    sendNotificationMessage(PUSHER_CHANNEL_WAREHOUSE, messageType.createRequest, { ...request.dataValues, book });
  });

  Request.afterUpdate(async (request) => {
    const book = await request.getBook();
    sendNotificationMessage(PUSHER_CHANNEL_WAREHOUSE, messageType.updateRequest, { ...request.dataValues, book });
  });

  return Request;
};

File: Project2/warehouse/interface/GUI/Request.cs
namespace @interface
{
    class Request
    {
        public int id { get; set; }
        public int quantity { get; set; }
        public string processedDate { get; set; }
        public Book book { get; set; }

        public Request()
        {
        }

        public Request(dynamic data, Book book)
        {
            id = (int) data.id;
            quantity = (int) data.quantity;
            processedDate = (string) data.processedDate;
            this.book = book;
        }

        public Request(int id, int quantity, string processedDate, Book book)
        {
            this.id = id;
            this.quantity = quantity;
            this.processedDate = processedDate;
            this.book = book;
        }
    }
}